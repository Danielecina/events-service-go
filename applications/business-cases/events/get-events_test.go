package businesscases

import (
	"fmt"
	"testing"

	entities "events-service-go/domains/entities/events"
	testutils "events-service-go/test-utils"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestGetAllEventsUseCase(t *testing.T) {
	t.Run("Execute_Success", func(t *testing.T) {
		expectedEvents := []entities.Event{
			{
				EventID:     "event-1",
				Name:        "Event 1",
				Description: "Desc 1",
				Location:    "Loc 1",
				UserID:      "user-1",
			},
			{
				EventID:     "event-2",
				Name:        "Event 2",
				Description: "Desc 2",
				Location:    "Loc 2",
				UserID:      "user-2",
			},
		}

		eventsRepositories := &testutils.MockEventRepositoryClient{
			GetAllMock: func(page int, limit int) ([]entities.Event, error) {
				return expectedEvents, nil
			},
		}

		useCase := NewGetEventsUseCase(eventsRepositories)

		response, err := useCase.Execute(1, 10)
		require.NoError(t, err)
		require.Len(t, response, 2)
		snaps.MatchSnapshot(t, response)
	})

	t.Run("Execute_Error", func(t *testing.T) {
		eventsRepositories := &testutils.MockEventRepositoryClient{
			GetAllMock: func(page int, limit int) ([]entities.Event, error) {
				return nil, fmt.Errorf("db error")
			},
		}

		useCase := NewGetEventsUseCase(eventsRepositories)
		got, err := useCase.Execute(1, 10)
		require.Error(t, err)
		require.Nil(t, got)
	})
}
