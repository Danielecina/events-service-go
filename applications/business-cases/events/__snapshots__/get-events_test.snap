
[TestGetAllEventsUseCase/Execute_Success - 1]
[]dto.GetEventResponse{
    {Name:"Event 1", Description:"Desc 1", Location:"Loc 1", UserID:"user-1", CreatedAt:"0001-01-01T00:00:00Z", EventID:"event-1"},
    {Name:"Event 2", Description:"Desc 2", Location:"Loc 2", UserID:"user-2", CreatedAt:"0001-01-01T00:00:00Z", EventID:"event-2"},
}
---
